src/components/PlaylistModal.jsx

```diff
@@ -44,7 +41,7 @@ export default function PlaylistModal({ video }) {
               onClick={() =>
                 videoInPlaylist(playlist)
                   ? toast.error("Video already exists in the playlist")
-                  : addToPlaylist(playlist._id, video, videoDispatch)
+                  : addToPlaylist(playlist._id, video, userToken, videoDispatch)
```

Comment:
suggestion :  import this usertoken in the addToPlaylist function file only

Reply:
That was what I did initially, but it wasn't capturing the userToken unless passed as a prop. I plan to look into the reason and change it later on, thank you!